src/components/books/bookAdd.js

```diff
@@ -1,36 +1,31 @@
 import React, { useState } from 'react';
 import { useDispatch } from 'react-redux';
 import { v4 as uuidv4 } from 'uuid';
-import { addBook } from '../../redux/books/booksSlice';
+import { postBook, addBook } from '../../redux/books/booksSlice';
 import './books.css';
 
 function BookAdd() {
   const dispatch = useDispatch();
   const [title, setTitle] = useState('');
   const [author, setAuthor] = useState('');
 
-  const book = {
-    item_id: '',
-    title: '',
-    author: '',
-  };
-
   const resetInput = () => {
     setTitle('');
     setAuthor('');
   };
 
   const submitHandler = (e) => {
     e.preventDefault();
+    if (!title.trim() || !author.trim()) return;
+
     if (title && author) {
-      dispatch(
-        addBook({
-          ...book,
-          item_id: uuidv4(),
-          title,
-          author,
-        }),
-      );
+      const bookData = {
+        item_id: uuidv4(),
+        title,
+        author,
+      };
+      dispatch(addBook(bookData));
+      dispatch(postBook(bookData));
       resetInput();
```

Comment:
- [ ] I was able to add and delete a book, but when I refreshed the page, I did not see my remaining books. I opened the dev tools to see the network tab, that's when I noticed a 400 error for a post request. Kindly take a look at the image below and make the changes.

![book](https://user-images.githubusercontent.com/53124548/222716143-073cc317-e3f5-4efe-823e-d5e346a83e36.png)